racadm.py

```diff
@@ -661,17 +666,141 @@ def arg_parse():
+            someDict[i+1] = {'Manufacturer': handle_data[key]['Manufacturer'], 'Size': handle_data[key]['Size'], 'Model': handle_data[key]['Model'], 'DeviceDescription': handle_data[key]['DeviceDescription']}
+        elif item == 'disk':
+            this_disk_size = int(handle_data[key]['SizeInBytes'].split()[0]) / 1024.0 / 1024.0 / 1024.0
+            all_disk_size += this_disk_size
+            someDict[i+1] = {'Manufacturer': handle_data[key]['Manufacturer'], 'Size': str(round(this_disk_size, 2)) + ' GB', 'Model': handle_data[key]['Model']}
+        elif item == 'raid':
+            someDict[i+1] = {'DeviceDescription': handle_data[key]['DeviceDescription']}
+
+    if item == 'cpu':
+        someDict['allNumberOfProcessorCores'] = allNumberOfProcessorCores
+        someDict['allNumberOfEnabledThreads'] = allNumberOfEnabledThreads
+    elif item == 'mem':
+        someDict['sum_mem'] = str(sum_mem / 1024) + ' GB'
+    elif item == 'disk':
+        if all_disk_size <= 1024:
+            someDict['all_disk_size'] = str(round(all_disk_size, 2)) + ' GB'
+        else:
+            someDict['all_disk_size'] = str(round(all_disk_size / 1024.0, 2)) + ' TB'
+
+    result[item] = someDict
+
+
+def check_ping(hostname):
+    response = os.system("ping -c 1 " + hostname + " 2>&1 > /dev/null")
+    # and then check the response...
+    if response == 0:
+        pingstatus = "Network Active"
+    else:
+        pingstatus = "Network Error"
+
+    return pingstatus
+
+
 def main():
     '''cli component of racadm'''
     args = arg_parse()
-    logging.basicConfig(level=logging.DEBUG)
-    racadm = Racadm(args.hostname, args.username, args.password, 
-            args.port )
-    print racadm.get_session_info()
+    logging.basicConfig(level=logging.ERROR)
+    if args.inputfile != '':
+        all_results = {}
+        with open(args.inputfile) as f:
+            for ip in f:
+                ip = ip.strip()
+                try:
+                    if check_ping(ip) == 'Network Active':
+                        racadm = Racadm(ip, args.username, args.password, args.port )
+                        result = analyse_result(racadm)
+                        all_results[ip] = result
+                        print json.dumps(result, indent=4, sort_keys=True)
+                        # print ip, result['sn'], result['system_model'], result['cpu']['allNumberOfEnabledThreads'], result['mem']['sum_mem'], result['disk']['all_disk_size']
+                    else:
+                        print ip, check_ping(ip)
+                except Exception:
+                        print ip, 'fetch info failed ...'
+        json.dump(all_results, open('./getIdracInfo.json', 'w'))
+        return
+
+    if args.hostname == 'localhost':
+        print 'python getIdracInfo.py -H ip -u root -p xxxxxxx || python getIdracInfo.py -f ip.list -u root -p xxxxxx '
+        return
+
+    if check_ping(args.hostname) == 'Network Active':
```

Comment:
connecting the host should not be reliant on icmp, can you please make this optinal.